roles/compute_build_nodes_cloud/tasks/main.yml

```diff
@@ -0,0 +1,95 @@
+---
+
+#   - name: print single node info
+#     debug:
+#       var: item.mac
+#     with_items: "{{ compute_nodes }}"
+#   - name: print single node info
+#     debug:
+#       var: item.vnfs
+#     with_items: "{{ compute_nodes }}"
+#
+#   - fail:
+#       msg: "Quick fail for test!"
+
+   - block:
+     
+     - name: Check if node is already in warewulf database
+       command: sudo wwsh node list
+       register: wwsh_node_list
+       ignore_errors: yes
+
+     - name: Add node(s) into warewulf database
+       command: wwsh -y node new {{ item.name }} --ipaddr={{ item.ip }} --hwaddr={{ item.mac }} -D {{ compute_private_nic }}
+       when: "'{{ item.name }}' not in wwsh_node_list.stdout"
+       with_items: "{{ compute_nodes }}"
+
+     - name: Register node IP and MAC address in warewulf database
+       command: sudo wwsh node set {{ item.name }} -y -I {{ item.ip }} --hwadd {{ item.mac }}
+       when: "'{{ item.name }}' in wwsh_node_list.stdout"
+       with_items: "{{ compute_nodes }}"
+
+     - name: set nodes bootloader
+       command: wwsh -y object modify -s bootloader=sda -t node {{ item.name }}
+       with_items: "{{ compute_nodes }}"
+       when: stateful_nodes == true
+  
+     - name: set nodes partitions
+       command: wwsh -y object modify -s diskpartition=sda -t node {{ item.name }}
+       with_items: "{{ compute_nodes }}"
+       when: stateful_nodes == true
+  
+     - name: format partitions
+       command: wwsh -y object modify -s diskformat=sda1,sda2,sda3 -t node {{ item.name }}
+       with_items: "{{ compute_nodes }}"
+       when: stateful_nodes == true
+  
+     - name: define filesystems
+       command: wwsh -y object modify -s filesystems="{{ sda1 }},{{ sda2 }},{{ sda3 }}" -t node {{ item.name }}
+       with_items: "{{ compute_nodes }}"
+       when: stateful_nodes == true
+  #" for vim
+
+     - name: set files to provision
+       command: wwsh -y provision set {{ item.name }} --bootstrap={{ build_kernel_ver }} --files=passwd,group,shadow,munge.key,slurm.conf,dynamic_hosts,network --bootlocal=exit
```

Comment:
please add `lmod.sh` and `lmod.csh` here. otherwise, compute nodes won't be able to use other module that we install in `/export`

Reply:
Done.